common/parse.go

```diff
@@ -57,6 +57,12 @@ func ParseResult(
 		nextPage = ""
 	}
 
+	if len(marshaledData) == 0 {
+		// Either a JSON array is empty or it was nil.
+		// For consistency return nil for missing records.
+		marshaledData = nil
```

Comment:
Why not []ReadResultRow{}, I believe most providers will return [].
Also are GetRecordsUnderJSONPath and GetOptionalRecordsUnderJSONPath not enough, as they take care of this.

Reply:
`GetRecordsUnderJSONPath` and `GetOptionalRecordsUnderJSONPath` will return either `[]` or `nil` and if excpectation is not met it would return an error. I would say these utilities are a bit low level. Maybe we should modify them to return empty initialized slice. Do you think this abstraction should rather live under `GetRecordsUnderJSONPath`?

We could return `[]ReadResultRow{}` in either case. I chose nil because it is a "default" golang value for a slice. Creating tests would require less fields.

From end user perspective: 
* looping over a nil works technically the same as empty slice
* there is one case, if user will try to modify and append a nil slice it would panic, but it should be a read only.

Reply:
I think there is no need to move it there, Here should be fine in my opinion, if someone uses another function for parsing records for any reason, he shouldn't re-write this. My thinking was these function will only return a nil response when there is an error, else it will be a list whether empty or with data. That's the reason I prefered [] rather nil.

Reply:
I decided to read some things about it, I would like to share:
https://stackoverflow.com/a/44305910/9628238 - check `Under the hood` section, very interesting.
https://gosamples.dev/empty-vs-nil-slice/ - article also describes behaviour. They don't mention the spread operator, but spreading nil slice works as "expected" and doesn't panic.

It seems that [] and nil are almost the same. Declaring nil is easier, thats what I found in unit tests (I even don't need to include the struct property, it is already nil).